error[E0119]: conflicting implementations of trait `bitflags::<unnamed>::convert::From<&_>` for type `types::to_sql::ToSqlOutput<'_>`

```text
# rustc --version
rustc 1.24.0-nightly (560a5da9f 2017-11-27)

# cargo --version 
cargo 0.24.0-nightly (abd137ad1 2017-11-12)
```

```toml
[lib]
name = "xxxx"
path = "src/lib.rs"
# crate-type = ["dylib"]

[build-dependencies]

[dependencies]
log = "0.3"
env_logger = "0.4"
futures = "0.1"
hyper = "0.11"
hyper-tls="0.1"
tokio-core = "0.1"
tokio-io = "0.1"
tokio-tls="0.1"
websocket = "0.20"
serde = "1.0"
serde_derive = "1.0"
serde_json = "1.0"
url= "1.4"
regex = "0.2"
uuid = { version = "0.4", features = ["serde", "v4"] }
encoding = "0.2"
chrono = { version = "0.4", features = ["serde", "rustc-serialize"] }
select = "0.4"

[dependencies.rusqlite]
version = "0.13"
features = ["bundled"]

[features]
default = []

```

Error message
```text
 Compiling websocket v0.20.2
   Compiling rusqlite v0.13.0
error[E0119]: conflicting implementations of trait `bitflags::<unnamed>::convert::From<&_>` for type `types::to_sql::ToSqlOutput<'_>`:
  --> $HOME/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/rusqlite-0.13.0/src/types/to_sql.rs:26:1
   |
18 | / impl<'a, T: ?Sized> From<&'a T> for ToSqlOutput<'a>
19 | |     where &'a T: Into<ValueRef<'a>>
20 | | {
21 | |     fn from(t: &'a T) -> Self {
22 | |         ToSqlOutput::Borrowed(t.into())
23 | |     }
24 | | }
   | |_- first implementation here
25 | 
26 | / impl<'a, T: Into<Value>> From<T> for ToSqlOutput<'a> {
27 | |     fn from(t: T) -> Self {
28 | |         ToSqlOutput::Owned(t.into())
29 | |     }
30 | | }
   | |_^ conflicting implementation for `types::to_sql::ToSqlOutput<'_>`

error: aborting due to previous error

error: Could not compile `rusqlite`.
warning: build failed, waiting for other jobs to finish...
error: build failed

```

--- a/src/types/to_sql.rs
+++ b/src/types/to_sql.rs
@@ -15,6 +15,8 @@ pub enum ToSqlOutput<'a> {
     ZeroBlob(i32),
 }
 
+// Generically allow any type that can be converted into a ValueRef
+// to be converted into a ToSqlOutput as well.
 impl<'a, T: ?Sized> From<&'a T> for ToSqlOutput<'a>
     where &'a T: Into<ValueRef<'a>>
 {
@@ -23,11 +25,31 @@ impl<'a, T: ?Sized> From<&'a T> for ToSqlOutput<'a>
     }
 }
 
-impl<'a, T: Into<Value>> From<T> for ToSqlOutput<'a> {
-    fn from(t: T) -> Self {
-        ToSqlOutput::Owned(t.into())
-    }
-}
+// We cannot also generically allow any type that can be converted
+// into a Value to be converted into a ToSqlOutput because of
+// coherence rules (https://github.com/rust-lang/rust/pull/46192),
+// so we'll manually implement it for all the types we know can
+// be converted into Values.
+macro_rules! from_value(
+    ($t:ty) => (
+        impl<'a> From<$t> for ToSqlOutput<'a> {
+            fn from(t: $t) -> Self { ToSqlOutput::Owned(t.into())}
+        }
+    )
+);
+from_value!(String);
+from_value!(Null);
+from_value!(bool);
+from_value!(i8);
+from_value!(i16);
+from_value!(i32);
+from_value!(i64);
+from_value!(isize);
+from_value!(u8);
+from_value!(u16);
+from_value!(u32);
+from_value!(f64);
+from_value!(Vec<u8>);
 
 impl<'a> ToSql for ToSqlOutput<'a> {
     fn to_sql(&self) -> Result<ToSqlOutput> {

